boefjes/boefjes/plugins/kat_nmap_udp/main.py

```diff
@@ -0,0 +1,60 @@
+from enum import Enum
+from ipaddress import IPv6Address, ip_address
+from os import getenv
+from typing import List, Optional, Tuple, Union
+
+import docker
+
+from boefjes.job_models import BoefjeMeta
+
+NMAP_IMAGE = "instrumentisto/nmap:latest"
+TOP_PORTS_MAX = 65535
+TOP_PORTS_DEFAULT = 250
+TOP_PORTS_MIN = 1
+
+
+def run_nmap(args: List[str]) -> str:
+    """Run Nmap in Docker."""
+    client = docker.from_env()
+    return client.containers.run(NMAP_IMAGE, args, remove=True).decode()
+
+
+class Protocol(Enum):
+    TCP = "tcp"
+    UDP = "udp"
+
+
+def build_nmap_arguments(host: str, protocol: Protocol, top_ports: Optional[int]) -> List[str]:
+    """Returns Nmap arguments to use based on protocol and top_ports for host."""
+    ip = ip_address(host)
+    args = ["--open", "-T4", "-Pn", "-r", "-v10", "-sV", "-sS" if protocol == Protocol.TCP else "-sU"]
```

Comment:
The Nmap UDP boefje is currently a literal copy of the Nmap TCP boefje, which is not great for maintainability.

I understand the desire to split the two boefjes, but I think this should be done using shared code instead of duplicating code into two boefjes.

Reply:
Fully agreed

> This solutions uses the same code on different places, which is a bad practice. Separate issue: https://github.com/minvws/nl-kat-coordination/issues/1468 https://github.com/minvws/nl-kat-coordination/issues/1343